index.js

```diff
@@ -72,12 +72,29 @@ function wrap(opts) {
 
 		onHash(input, metadata, hash);
 
-		try {
-			return fs.readFileSync(cachedPath, encoding);
-		} catch (error) {
-			const result = transform(input, metadata, hash);
-			writeFileAtomic.sync(cachedPath, result, {encoding});
-			return result;
+		let result;
+		let retry = 0;
+		/* eslint-disable-next-line no-constant-condition */
+		while (true) {
+			try {
+				return fs.readFileSync(cachedPath, encoding);
+			} catch (error) {
+				if (!result) {
+					result = transform(input, metadata, hash);
+				}
+
+				try {
+					writeFileAtomic.sync(cachedPath, result, {encoding});
+					return result;
+				} catch (error) {
```

Comment:
Forgive me if I'm wrong, but isn't this a naming collision with the error object from the previous catch block?

Reply:
The variable is scoped, so `error` from the inner catch is what we throw on line 94.  This is just habit I always use `catch (error) { ... }` except when I need the outer error from an inner catch.  I can see how this could be confusing I can change the outer scope to `readError` if you want.

Reply:
I mean the error from the previous scope is also never used so it would be nice to have the new `catch {}` syntax

Reply:
As much as I'd love to eliminate node.js 8 and 10 right now ... :smile: